idaes/models_extra/power_generation/costing/power_plant_capcost.py

```diff
@@ -0,0 +1,1995 @@
+
+        blk.total_plant_cost_eq = Constraint(cost_accounts, rule=total_plant_cost_rule)
+
+        # rule for sum of BEC
+        def BEC_sum_rule(blk):
+            return sum(blk.bare_erected_cost[i] for i in cost_accounts)
+
+        blk.bare_erected_cost_sum = Expression(rule=BEC_sum_rule)
+
+        # rule for sum of TPC
+        def TPC_sum_rule(blk):
+            return sum(blk.total_plant_cost[i] for i in cost_accounts)
+
+        blk.total_plant_cost_sum = Expression(rule=TPC_sum_rule)
+
+        # add variable and constraint scaling
+        for i in cost_accounts:
+            iscale.set_scaling_factor(blk.bare_erected_cost[i], 1)
+            iscale.set_scaling_factor(blk.total_plant_cost[i], 1)
+            iscale.constraint_scaling_transform(
+                blk.bare_erected_cost_eq[i], 1e-3, overwrite=False
+            )
+            iscale.constraint_scaling_transform(
+                blk.total_plant_cost_eq[i], 1, overwrite=False
+            )
+
+    # -----------------------------------------------------------------------------
+    # Supercritical CO2 Costing Library
+    # -----------------------------------------------------------------------------
+    def get_sCO2_unit_cost(
+        self,
+        equipment,
+        scaled_param,
+        temp_C=None,
+        n_equip=1,
+        CE_index_year="2018",
+        custom_accounts=None,
+    ):
+        """
+        Args:
+            self: pyomo Block where constraints will be made
+            unit_name: the name of the SCO2 equipment to cost
+            scaling_param: the scaling parameter (in appropriate units) for the
+                elected equipment
+            temp: the maximum temperature of the equipment. Not all types of
+                equipment use a temperature correction factor, so it is optional
+            n_equip: the number of pieces of equipment to cost
+
+        Cost is in M$
+        """
+        # check to see if a costing block already exists
+        if hasattr(self, "costing"):
+            raise AttributeError(
+                "{} already has an attribute costing. "
+                "Check that you are not calling get_costing"
+                " twice on the same model".format(self.name)
+            )
+
+        # create a costing Block
+        self.costing = Block()
+        self.costing.library = "sCO2"
+        self.costing.equipment = equipment
+
+        try:
+            CE_index_units = getattr(pyunits, "MUSD_" + CE_index_year)
+        except AttributeError:
+            raise Exception(
+                "CE_index_year %s is not a valid currency base option. "
+                "Valid CE index options include CE500, CE394 and years from "
+                "1990 to 2020." % (CE_index_year)
+            )
+
+        param_dict = sCO2_costing_params[equipment]
+
+        if custom_accounts is not None:
+            # adding custom params to the dictionary
+            for key, value in custom_accounts.items():
```

Comment:
Consider to change `value` for `val` in this loop as the method value is being imported from pyomo and this may cause some confusion